### PR TITLE
Add isRightToLeft function and some scripts

### PR DESCRIPTION
**Pull request by SafaAlfulaij**
This adds a function to decide if the subtitle line is RTL or not, which makes it possible for scripts to check that and do some post-processing due to the stupidness of ASS engines (libASS and vsFilter) as well as the normal SRT format that doesn't support that, unfortunately.
I didn't check other formats supported really.

### CODE REVIEW

File: src/core/subtitleline.cpp [hunk 1]
```diff
@@ -1044,6 +1044,12 @@ SubtitleLine::check(int errorFlagsToCheck, int minDurationMsecs, int maxDuration
 	return lineErrorFlags;
 }
 
+bool
+SubtitleLine::isRightToLeft()
```
Comment (Martchus) [inline src/core/subtitleline.cpp]: Sure this can not be const?

File: src/scripting/scripting_subtitleline.cpp [hunk 2]
```diff
@@ -418,4 +418,8 @@ Scripting::SubtitleLine::check(int errorFlagsToCheck, int minDuration, int maxDu
 	return m_backend->check(errorFlagsToCheck, minDuration, maxDuration, minDurationPerChar, maxDurationPerChar, maxChars, maxLines, update);
 }
 
-
+bool
+Scripting::SubtitleLine::isRightToLeft()
```
Comment (Martchus) [inline src/scripting/scripting_subtitleline.cpp]: Then this might be const, too.

**Comment by SafaAlfulaij** [comment]
To be honest, I don't know the deep meanings of `const` stuff, just took a look on the existing code with some reading of that concept.

**Comment by Martchus** [comment]
@SafaAlfulaij But you did it right. The `const` will make the target of the `this` pointer inside the methods `const`.

**Comment by maxrd2** [comment]
Thank you, it looks good but am wondering if it would be better to add RTL support directly into subtitlecomposer code. That script just adds RTL marks on beginning of each line when QString::isRightToLeft() returns true? Maybe it would be better if I add logic to include those marks immediately when loading/saving subtitles?
When/why are those needed?

**Comment by maxrd2** [comment]
Have merged this for now. Will look into adding better support for RTL text to the application itself in the future.

**Pull request merged**